backend/src/test/java/ai/verta/modeldb/BranchTest.java

```diff
@@ -693,15 +713,23 @@ public void branchTest() {
         2,
         listBranchCommitsResponse.getCommitsCount());
 
-    Collections.reverse(commitShaList);
     commitShaList.forEach(
         commitSha -> {
           DeleteCommitRequest deleteCommitRequest =
               DeleteCommitRequest.newBuilder()
                   .setRepositoryId(RepositoryIdentification.newBuilder().setRepoId(id).build())
                   .setCommitSha(commitSha)
                   .build();
-          versioningServiceBlockingStub.deleteCommit(deleteCommitRequest);
+          if (commitShaList.get(3).equals(commitSha)) {
```

Comment:
add either inline comment or a block comment for the function on what condition you are testing. Nothing fancy, even the current PR title can work

Reply:
Added